Sources/Utility/ArgumentParser.swift

```diff
@@ -82,64 +81,61 @@ public protocol ArgumentKind {
     ///
     /// Call methods on the passed parser to manipulate parser as needed.
     init(parser: inout ArgumentParserProtocol) throws
-
-    /// This will be called for positional arguments with the value discovered.
-    init?(arg: String)
-
+    
+    /// Failable initializer to convert from String.
+    init?(argument: String)
+    
     /// Type of shell completion to provide for this argument.
     static var completion: ShellCompletion { get }
 }
 
-// MARK: - ArgumentKind conformance for common types
-
-extension String: ArgumentKind {
-    public init?(arg: String) {
-        self = arg
+extension ArgumentKind {
+    public static func parse(argument: String) throws -> Self {
```

Comment:
Hmmm, it if is done separately it would have to be done beforehand, because this PR depends on this change to make the code in `ParsingStatregy` very DRY. Do you me to get it through first?